src/renderer/viz/expressions/buckets.js

```diff
@@ -150,6 +150,13 @@ export default class Buckets extends BaseExpression {
             }
         });
 
+        if (this.input.type === 'category') {
+            const property = metadata.properties[this.input.propertyName];
+            this._numDatasetCategories = property
+                ? property.categories.length
+                : this.list.elems.length - 1;
```

Comment:
When is this else condition reached?

Reply:
This is an specific use case where the buckets input doesn't depend on a feature, and you can use a string as an input. This is used for getting a value. We've tests with this use case in the `ramp.test.js` file, for example:

```js
ramp(buckets('Madrid', ['Pontevedra', 'Zaragoza', 'Cordoba', 'Alicante', 'Murcia']), palettes.PRISM, red);
```